tests/std/tests/P2286R8_text_formatting_range_string/test.cpp

```diff
@@ -0,0 +1,563 @@
+    }
+
+    template <class CharT, size_t N>
+    basic_string<CharT> get_colons() {
+        return basic_string<CharT>(N, CharT(':'));
+    }
+
+    template <class CharT, /*format_types types,*/ size_t N>
+    vector<basic_string<CharT>> fmt_invalid_types(string_view valid) {
+        vector<basic_string<CharT>> result;
+        if constexpr (is_permissive_v<CharT>) {
+            for (const char c : get_format_types()) {
+                if (valid.find(c) == string_view::npos) {
+                    result.push_back(format(SV("{{{}{}}}"), get_colons<CharT, N>(), c));
+                }
+            }
+        } else {
+            // ranges::to is not available in C++20.
+            ranges::copy(get_format_types() | views::filter([&](char type) {
+                return valid.find(type) == string_view::npos;
+            }) | views::transform([&](char type) { return format(SV("{{{}{}}}"), get_colons<CharT, N>(), type); }),
+                back_inserter(result));
+        }
+        return result;
+    }
+} // namespace detail
+
+// Creates format string for the invalid types.
+//
+// valid contains a list of types that are valid.
+//
+// The return value is a collection of basic_strings, instead of
+// basic_string_views since the values are temporaries.
+template <class CharT>
+vector<basic_string<CharT>> fmt_invalid_types(string_view valid) {
+    return detail::fmt_invalid_types<CharT, 1>(valid);
+}
+
+// Like fmt_invalid_types but when the format spec is for an underlying formatter.
+template <class CharT>
+vector<basic_string<CharT>> fmt_invalid_nested_types(string_view valid) {
+    return detail::fmt_invalid_types<CharT, 2>(valid);
+}
+
+template <class T>
+struct context {};
+
+template <>
+struct context<char> {
+    using type = format_context;
+};
+
+template <>
+struct context<wchar_t> {
+    using type = wformat_context;
+};
+
+template <class T>
+using context_t = context<T>::type;
+
+
+//
+// Types
+//
+
+template <class Container>
+class test_range_format_string {
+public:
+    explicit test_range_format_string(Container str) : str_(move(str)) {}
+
+    typename Container::const_iterator begin() const {
+        return str_.begin();
+    }
+    typename Container::const_iterator end() const {
+        return str_.end();
+    }
+
+private:
+    Container str_;
+};
+
+template <class Container>
+constexpr range_format std::format_kind<test_range_format_string<Container>> = range_format::string;
+
+template <class Container>
+class test_range_format_debug_string {
+public:
+    explicit test_range_format_debug_string(Container str) : str_(move(str)) {}
+
+    typename Container::const_iterator begin() const {
+        return str_.begin();
+    }
+    typename Container::const_iterator end() const {
+        return str_.end();
+    }
+
+private:
+    Container str_;
+};
+
+template <class Container>
+constexpr range_format format_kind<test_range_format_debug_string<Container>> = range_format::debug_string;

```

Comment:
`std::` was removed from this partial specialization, but retained on 129, I think we should be consistent.